lib/jormungandr/test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs

```diff
@@ -261,7 +262,27 @@ spec = do
     describe "White-box error path tests" $
         around startNode $ do
 
-        it "can't get account state for account that does not exist" $
+        it "getAccountState: Invalid url path" $
+            \(_, url) -> do
+                manager <- newManager defaultManagerSettings
+                let wrongUrl = url { baseUrlPath = "/not-valid-prefix" }
+                let client = Jormungandr.mkJormungandrClient manager wrongUrl
+                res <- runExceptT $
+                    Jormungandr.getAccountState client testAccountId
+                res `shouldBe` Left
+                    (ErrGetAccountStateAccountNotFound testAccountId)
+
+        it "getAccountState: ErrGetAccountStateNetworkUnreachable" $
+            \(_, url) -> do
+                manager <- newManager defaultManagerSettings
+                let wrongPort = (baseUrlPort url) + 1
+                let wrongUrl = url { baseUrlPort = wrongPort }
+                let client = Jormungandr.mkJormungandrClient manager wrongUrl
+                Left (ErrGetAccountStateNetworkUnreachable err) <- runExceptT $
+                    Jormungandr.getAccountState client testAccountId
+                (show err) `shouldContain` "Connection refused"
```

Comment:
I'd suggest to follow an approach similar to what I did in #1054 (have a separate helper function that we can re-use for all `GET` endpoints of the Jörmungandr Client) :+1:

Reply:
yeah, I see now :)

Reply:
I give up :(
I've spent too much time trying to rework like:

```haskell
testNetworkUnrechable
    :: (Show err, Eq err, Show result, Eq result)
    => String
    -> (forall m. JormungandrClient m -> resourceId -> ExceptT err m result)
    -> resourceId
    -> (Jormungandr.ErrNetworkUnavailable -> err)
    -> SpecWith (whatever, BaseUrl)
testNetworkUnrechable fn endpoint resourceId netErr =
    it (fn <> ": network unreachable") $ \(_, url) -> do
        manager <- newManager defaultManagerSettings
        let url' = url { baseUrlPort = baseUrlPort url + 5 }
        let client = Jormungandr.mkJormungandrClient manager url'
        res <- runExceptT $ endpoint client resourceId
        -- res `shouldThrow` \case
        --         Left (netErr _) -> True
        --         _ -> False
        Left (netErr msg)
                <- runExceptT $ endpoint client resourceId
        (show msg) `shouldContain` "Connection refused"
```

but always getting strange error that I was not able to resolve:

```
  Parse error in pattern: netErr                 
  Possibly caused by a missing 'do'? 
```

So just adding those tests as they were... :(

Reply:
If you look at the type of `shouldThrow`:

```hs
shouldThrow :: (HasCallStack, Exception e) => IO a -> Selector e -> Expectation
```

Where, `Selector` is:

```hs
type Selector e = e -> Bool
```

Said differently, it is polymorphic in `e` and `e` has to be an instance of `Exception`. The pattern-match should reflect that and actually "make a choice" about what `e` is. This allows to only run a predicate on a very specific `Exception`. This is why in the previous PR, I moved the selector into a separate function with a clear type annotation for to specialize the `e` into a `ErrUnexpectedNetworkFailure`

https://github.com/input-output-hk/cardano-wallet/blob/master/lib/jormungandr/test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs#L483-L485

In your example above, two things at least need fix:

1. The pattern match should not be on an `Either` Right/Left, but on an exception data-type directly
2. Either pattern match on the exception constructor, or, give a clear type annotation of the exception you're looking into to allow the type-system to figure out what should be caught.

Note that I am not sure that the lambda case syntax `\case` would works here. It probably will, but it may good to move the assertion to a separate function.

Reply:
Ok, reworked and ended up adding a bit more tests (for other JormungandrClient endpoints also).